src/libproc_macro/lib.rs

```diff
@@ -1168,6 +1169,50 @@ impl Literal {
     pub fn set_span(&mut self, span: Span) {
         self.span = span;
     }
+
+    /// Returns a `Span` that is a subset of `self.span()` containing only the
+    /// source bytes in range `range`. Returns `None` if the would-be trimmed
+    /// span is outside the bounds of `self`.
+    // FIXME(SergioBenitez): check that the byte range starts and ends at a
+    // UTF-8 boundary of the source. otherwise, it's likely that a panic will
+    // occur elsewhere when the source text is printed.
+    // FIXME(SergioBenitez): there is no way for the user to know what
+    // `self.span()` actually maps to, so this method can currently only be
+    // called blindly. For example, `to_string()` for the character 'c' returns
+    // "'\u{63}'"; there is no way for the user to know whether the source text
+    // was 'c' or whether it was '\u{63}'.
```

Comment:
I don't think this is correct. `Literal::to_string` should show *the exact* source contents, if the literal isn't synthetic.

Reply:
This is correct.

Reply:
If it's synthetic then the span doesn't matter, I was talking about `Literal`s from the source. But also, we shouldn't be using `escape_unicode` like that.